software/tracksight/frontend/src/app/components/dropdown_menu.tsx

```diff
@@ -1,52 +1,63 @@
-import { useState, useEffect, React } from 'react';
+import { useState, useEffect } from 'react'; // Removed extra React import
 import { DownOutlined, SmileOutlined } from '@ant-design/icons';
-import type { MenuProps } from 'antd';
 import { Dropdown, Space } from 'antd';
-import WebSocketComponent from './web_socket';
-
 
 const DropdownMenu = (props) => {
-    const [data, setData] = useState([]) // Format is list of signals 
-    const [items, setItems] = useState<MenuProps['items']>([]);
-    const noItems: [] = [{
-        key: "1",
-        label: "No Avaliable Signals"
-    }];
-
-    // Request available signals through socket on render
-    props.socket.emit("available_signals", {"ids" : []});
+    const [items, setItems] = useState([]);
+    const [visible, setVisible] = useState(false);
+    const [selectedSignals, setSelectedSignals] = useState([]);
 
-    useEffect(() => {
-        if (data.length > 0) {
-            const res = []
-            for (let i = 0; i < data.length; i++) {
-                res.push({
-                    key: i.toString(),
-                    label: (
-                        data[i]
-                    )
-                })
-            }
-            setItems(res)
+    const handleSignalClick = (signalName) => {
+        const newSelectedSignals = [...selectedSignals];
+        const index = newSelectedSignals.indexOf(signalName);
+        if (index === -1) {
+            newSelectedSignals.push(signalName);
         } else {
-            setItems(noItems)
+            newSelectedSignals.splice(index, 1);
         }
-    }, [data]);
+        setSelectedSignals(newSelectedSignals);
+        props.setSignal(newSelectedSignals);
+    };
+
+    useEffect(() => {
+        const updatedItems = props.signals.map((signalName, index) => ({
+            key: index.toString(),
+            label: (
+                <p>
+                    <input 
+                        type="checkbox" 
+                        checked={selectedSignals.includes(signalName)}
+                        onChange={() => handleSignalClick(signalName)} // Added onChange handler
```

Comment:
Instead of having duplicate handleSignalClick are you able to put it once on the surrounding tag?(p right now, Space in the future)